koji/collection-mash-split.py

```diff
@@ -324,6 +341,9 @@ def handle_comps(self, whole_path, tmp_path, split_path, mash_config, arches, co
                 source = os.path.join(whole_path, mash_config, arch, "os", "Packages")
                 copied.update(self.copyout(comps_pkg_names, source, tmp_target))
                 self.createrepo(tmp_target, comps, checksum=checksum)
+                if modulemd_yaml:
+                    modulemd = self.get_from_git(gitloc, modulemd_baseloc, modulemd_yaml, tmp_target)
```

Comment:
I wonder if this should be unpacked to `tmp_target` or `comps_path` or something else. `get_from_git` returns a full path, so we don't care *where* on disk the file is, as long as we can read it, and we don't need the intermediate version in the resulting repo which `tmp_target` technically represents.

Reply:
It would save the whole `git archive`, though that probably isn't very expensive since it's just retrieving it for a single filename. That said, `git show remotes/origin/%s:%s` and return that would be more efficient. In `handle_extras` it would save writing it to disk altogether and in this case it would avoid writing it to disk once.

Reply:
also works with the comps usecase, I think

Reply:
mm, no, it wants an file-like object, or a path here.

StringIO here I come?

Reply:
I'd prefer moving this to a next PR, as an overall improvement of `get_from_git`, mostly because I'd prefer this PR to be as simple as possible ;)

Reply:
#1702 is a draft for that effort